src/Nox.Types/Types/StreetAddress/StreetAddress.cs

```diff
@@ -1,10 +1,213 @@
+using System.Linq;
+using System.Text.RegularExpressions;
+
 namespace Nox.Types;
 
-    /// <summary>
-    /// Represents a Nox <see cref="StreetAddress"/> type and value object.
-    /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class StreetAddress : ValueObject<(string Line1,string Line2, string Line3,string ZipCode,string City, CountryCode2 CountryCode2), StreetAddress>
+/// <summary>
+/// Represents a Nox <see cref="StreetAddress"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
+public sealed class StreetAddress : ValueObject<
+    (int StreetNumber,
+    string AddressLine1,
+    string AddressLine2,
+    string Route,
+    string Locality,
+    string Neighborhood,
+    string AdministrativeArea1,
+    string AdministrativeArea2,
+    string PostalCode,
+    CountryCode2 CountryId), StreetAddress>
+{
+    private readonly Regex _postalCodeRegex = new("^\\d{5}(?:[-\\s]\\d{4})?$");
+
+    public int StreetNumber
+    {
```

Comment:
for this type the get/set are too cumbersome
maybe we should use a interna class(POCO) instead of a named tuple, what do you think?